source/common/tracing/dynamic_opentracing_driver_impl.cc

```diff
@@ -0,0 +1,38 @@
+#include "common/tracing/dynamic_opentracing_driver_impl.h"
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Tracing {
+
+DynamicOpenTracingDriver::DynamicOpenTracingDriver(Stats::Store& stats, const std::string& library,
+                                                   const std::string& tracer_config)
+    : OpenTracingDriver{stats} {
+  std::string error_message;
+  opentracing::expected<opentracing::DynamicTracingLibraryHandle> library_handle_maybe =
+      opentracing::DynamicallyLoadTracingLibrary(library.c_str(), error_message);
+  if (!library_handle_maybe) {
+    if (error_message.empty()) {
+      throw EnvoyException{fmt::format("{}", library_handle_maybe.error().message())};
```

Comment:
Is it possible to test these cases, here and below?

Reply:
I simplified so that there's only a single code path now that should get hit in the testing.